manifests/05-operator.yaml

```
@@ -13,6 +13,8 @@ spec:
       labels:
         name: console-operator
     spec:
+      nodeSelector:
+        "node-role.kubernetes.io/master": ""
```

NIT: I think it should be
```
node-role.kubernetes.io/master: ''
```

---

Yeah, it's a weird annotation, has no value.  But its consistent with elsewhere. 🤷‍♂️

---

> Just out of curiosity, why the value is empty ?

You have to match the label on the node, and that's what set for masters.

```yaml
  labels:
    beta.kubernetes.io/arch: amd64
    beta.kubernetes.io/instance-type: m4.large
    beta.kubernetes.io/os: linux
    failure-domain.beta.kubernetes.io/region: us-east-2
    failure-domain.beta.kubernetes.io/zone: us-east-2a
    kubernetes.io/hostname: ip-10-0-13-185
    node-role.kubernetes.io/master: ''
```

---

We might want to just restrict the operator to linux if it doesn't matter if it runs on the master.

```suggestion
        beta.kubernetes.io/os: linux
```

---

> You have to match the label on the node, and that's what set for masters.

Have spoken to @mfojtik regarding the additional labels, and was told that in order to run the operator on one of the masters the given `nodeSelector` should be enough. 
Regarding specifying the linux machine, that shouldn't be necessary since every master is running on linux machine.

---

Yeah, I'm not saying add both. I'm saying use that instead of the master node selector.

Let me follow up on this today. I'm not sure what the recommendation is for operators.

---

Hmm... Yeah seems like we can loosen up to any Linux for the operator, so long as the console itself is on masters. Not a big deal if it gets destroyed and recreated.

---

> From what I was told the recommendation is just the node selector, although there might be more to it.

Right, it's still a node selector. Just a different value if we don't require the operator to run on the masters.

---

Not sure why we don't would want to run operator on master.

---

I checked with @derekwaynecarr, and he recommended letting the operator run on any linux node by default. We'll need to add a config option to let admins configure the node selector later, but for now, let's go with:

```yaml
nodeSelector:
  beta.kubernetes.io/os: linux
```

---

I just realized that it's not possible to make operator node selector configurable since this is a static manifest that the CVO applies. I raised the issue with Derek.

---

OK, change of plans. We've been asked to run the operator on the master :)